node/src/components/block_proposer.rs

```diff
@@ -383,6 +395,12 @@ impl BlockProposerReady {
 
         let mut transfers = Vec::new();
         let mut wasm_deploys = Vec::new();
+        let mut block_gas_running_total = 0u64;
+
+        // TODO: is this possible to check this in BlockProposer:
```

Comment:
@goral09 and @afck can I please get a sanity check here? How would we check/enforce the size of the proposed block?

Reply:
I agree that this requires serializing the block and all its deploys, and summing up the sizes. Obviously we don't have the serialized block here yet, so it's only about the length of the deploy hash list plus the sizes of the deploys (plus maybe some constant for the rest of the block?). We don't have to serialize the former since all deploy hashes have the same size. How to do the latter I don't know; maybe this should happen in another component (e.g. store each deploy's size when we receive it, before deserializing it)?

Reply:
From our sidebar on this, the end result is that we can calculate it from the ExecutableDeployItem and hashes.